### PR TITLE
[xcode15] Add support for Xcode 15 and corresponding OS versions.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: Make.config [hunk 1]
```diff
@@ -200,9 +200,9 @@ MACCATALYST_NUGET_VERSION_NO_METADATA=$(MACCATALYST_NUGET_VERSION)$(NUGET_PREREL
 MACCATALYST_NUGET_VERSION_FULL=$(MACCATALYST_NUGET_VERSION_NO_METADATA)+$(NUGET_BUILD_METADATA)
 
 # Xcode version should have both a major and a minor version (even if the minor version is 0)
-XCODE_VERSION=14.3.1
-XCODE_URL=https://dl.internalx.com/internal-files/xcodes/Xcode_14.3.1.xip
-XCODE_DEVELOPER_ROOT=/Applications/Xcode_14.3.1.app/Contents/Developer
+XCODE_VERSION=15.0
+XCODE_URL=https://dl.internalx.com/internal-files/xcodes/Xcode_15_beta_4.xip
```
Comment (filipnavara) [inline Make.config:204]: Not going straight for beta 5? 😅

Comment (mandel-macaque) [reply Make.config:204]: Let me get Beta 4 in a decent place first 😅

File: msbuild/Xamarin.MacDev.Tasks/Tasks/LinkNativeCodeTaskBase.cs [hunk 2]
```diff
@@ -75,6 +75,9 @@ bool ExecuteUnsafe ()
 
 			var arguments = new List<string> ();
 			arguments.Add ("clang++");
+			// check if needs to be removed: https://github.com/xamarin/xamarin-macios/issues/18556
+			arguments.Add ("-Xlinker");
+			arguments.Add ("-ld64");
```
Comment (filipnavara) [inline msbuild/Xamarin.MacDev.Tasks/Tasks/LinkNativeCodeTaskBase.cs:80]: In beta 5 this was renamed to `ld_classic`. The new linker is still as broken as ever.

Comment (mandel-macaque) [reply msbuild/Xamarin.MacDev.Tasks/Tasks/LinkNativeCodeTaskBase.cs:80]: Good lord apple

**Comment by mandel-macaque** [comment]
something funky happened, you can find the new pr here: https://github.com/xamarin/xamarin-macios/pull/18612

Same fork branch

**Pull request merged**